enhancements/authentication/separate-oauth-resources.md

```diff
@@ -0,0 +1,142 @@
+
+## Motivation
+
+The OpenShift OAuth server requires several openshift API resources in order to function.
+Because the OAuth server enables access to the kube-apiserver, it needs to be able to run separately from the existing
+set of API resources that are focused on enabling a developer workflow.
+This narrowing of scope expands the deployment options for the OpenShift OAuth server.
+
+### Goals
+
+1. Make it possible to run the OpenShift OAuth server without installing all the developer focused API Resources.
+
+### Non-Goals
+
+1. Move routes out of the OpenShift API Server.
+
+## Proposal
+
+1. Create a new repo called `openshift-oauth-apiserver` that produces a binary called `openshift-oauth-apiserver`.
+2. Add the following API groups to the new repo
+   1. authorization  - most of this API group is deprecated.  It will sleep here until we're ready to retire it.
+   2. oauth - core oauth types
+   3. project - intersection of authorization and namespaces.  It is heavily dependent on the way our scopes work.
+   4. quota - intersection of users and namespaces
+   5. security - SCC
+   6. user - core types used by oauth and authentication
+3. Update the authentication operator to install the new apiserver, but not wire the apiservice.
+4. Add the code to the openshift-apiserver-operator tolerate apiservice management by the authentication operator
+(indicated by annotation) if the authentication operator is available at the 4.n version.  This code will need to be 
+backported to 4.n-1.
+5. Update the authentication operator to wire the apiservices.
+6. Remove the registration from the openshift-apiserver-operator.
+7. Remove the serving code from the openshift-apiserver.
+
+If we stop at any any intermediate step, we are still safe to ship.
+
+### User Stories [optional]
+
+#### Story 1
+
+#### Story 2
+
+### Implementation Details/Notes/Constraints [optional]
+
+What are the caveats to the implementation? What are some important details that
+didn't come across above. Go in to as much detail as necessary here. This might
+be a good place to talk about core concepts and how they releate.
+
+### Risks and Mitigations
+
+What are the risks of this proposal and how do we mitigate. Think broadly. For
+example, consider both security and how this will impact the larger OKD
+ecosystem.
+
+How will security be reviewed and by whom? How will UX be reviewed and by whom?
+
+Consider including folks that also work outside your immediate sub-project.
+
+## Design Details
+
+### Test Plan
+
+If the system works, we did it.
+
+##### Removing a deprecated feature
+
+See upgrade/downgrade.
+
+### Upgrade / Downgrade Strategy
+
+This is an interesting situation.  We want to hand control of the API over to a new process.  This is possible to do,
+however, to do it in a single release requires coordination.
+
+1. Add code in 4.(n-1) to the openshift-apiserver-operator that prevents it from managing an apiservice if it is "claimed"
```

Comment:
What happens when a user of `4.(n-1).z`, which does not have this backport, upgrades directly to `4.n`? Is path of skipping versions supported or does the cluster need to go through all the `4."..."` upgrades first?

Reply:
> What happens when a user of `4.(n-1).z`, which does not have this backport, upgrades directly to `4.n`? Is path of skipping versions supported or does the cluster need to go through all the `4."..."` upgrades first?

As before, we will only draw edges from late 4.(n-1) to 4.n